repos/base-hw/src/bootstrap/spec/arm/cpu.cc

```diff
@@ -27,12 +27,10 @@ void Bootstrap::Cpu::enable_mmu_and_caches(Genode::addr_t table)
 	Ttbcr::write(1);
 
 	Ttbr::access_t ttbr = Ttbr::Ba::masked(table);
-	Ttbr::Rgn::set(ttbr, Ttbr::CACHEABLE);
-	if (Mpidr::read()) { /* check for SMP system */
-		Ttbr::Irgn::set(ttbr, Ttbr::CACHEABLE);
-		Ttbr::S::set(ttbr, 1);
-	} else
-		Ttbr::C::set(ttbr, 1);
+	Ttbr::Rgn::set(ttbr, Ttbr::BACK_NOALLOCATE);
+	Ttbr::Irgn::set(ttbr, Ttbr::BACK_NOALLOCATE);
+	Ttbr::S::set(ttbr, 1);
+	Ttbr::C::set(ttbr, 1);
```

Comment:
Note that this only affects the caching of page table memory. The page table entries still specify a write-back allocate policy.

At least the IRGN must be set to write-back no write-allocate, certainly because the SCU does not maintain coherence between L1 caches. I was not able to bring up the system properly with `Ttbr::BACK_ALLOCATE`. However, I re-checked that it still works with `Ttbr::BACK_ALLOCATE` for RGN.

Reply:
By the way: IRGN can also be set to write-through, which is probably the more intuitive and safer policy in this case.

Reply:
I must actually correct myself. I was misled by a bug in the Lauterbach tool, which misinterpreted the IRGN bitfield due to which I wrongly modified the implementation of `Ttbr::Irgn`. I reverted this change and re-checked with what settings the Zynq-7000 successfully boots into SMP. In summary, only write-back write-allocate works (according to the CCSIDR there is actually no write-through support on level 1). I added a check for the multiprocessing extension in order to set the IRGN, RGN and S bits conditionally.